plugins/modules/web_app_pool.ps1

```diff
@@ -0,0 +1,347 @@
+#!powershell
+
+# Copyright: (c) 2015, Henrik Wallström <henrik@wallstroms.nu>
+# Copyright: (c) 2017, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$ErrorActionPreference = 'Stop'
+
+$spec = @{
+    options = @{
+        name = @{
+            required = $true
+            type = "str"
+        }
+        state = @{
+            type = "str"
+            default = "present"
+            choices = @("absent", "present", "restarted", "started", "stopped")
+        }
+        attributes = @{
+            type = "dict"
+            # default empty dictionary to ensure it's always a dictionary
+            default = @{}
+        }
+    }
+    supports_check_mode = $true
+}
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$name = $module.Params.name
+$state = $module.Params.state
+$check_mode = $module.CheckMode
+$module.Result.info = @{
+    name = $name
+    state = $state
+    attributes = @{}
+    cpu = @{}
+    failure = @{}
+    processModel = @{}
+    recycling = @{
+        periodicRestart = @{}
+    }
+}
+$module.Result.changed = $false
+
+# Stores the free form attributes for the module
+$attributes = $module.Params.attributes
+$module.Result.attributes = $attributes
```

Comment:
Ditto here, we should avoid returning what we already know, in this case the `attributes` are just the input values and they are automatically added by `Ansible.Basic` when running with a verbosity of `-vvv` or higher which is common Ansible practice.